c10/core/MemoryFormat.cpp

```diff
@@ -0,0 +1,15 @@
+#include <c10/core/MemoryFormat.h>
+namespace c10 {
+namespace {
+thread_local bool memory_format_propagation_enabled = false;
```

Comment:
@gchanan has stated that it is too difficult to write code that works under both settings of this toggle, and thus we should not release with this toggle available to users. I have an alternate proposal, which I have written up at #24261 for ensuring this toggle only affects user code, and not internal code.